oars/models/enrollment/enrollments_by_day.sql

```diff
@@ -1,16 +1,27 @@
--- one record per learner per day with enrollment status
+-- one record per learner, org, course, and day with
+-- enrollment status as of 11:59 UTC of that day
+--
+-- a range of dates is calculated by converting DateTime fields
+-- to Unix timestamps, then generating a range of Unix timestamps,
+-- creating one record per timestamp (via arrayJoin), and then converting
+-- the timestamps back into dates
+
 
 select
-  d.date_day,
-  e.actor_id,
-  e.object_id,
-  e.course_id,
-  e.org,
-  e.event_type as enrollment_status,
-  e.enrollment_mode
+  date(fromUnixTimestamp(
+    arrayJoin(
+      range(
+        toUnixTimestamp(window_start),
+        toUnixTimestamp(window_end),
+        24 * 60 * 60
```

Comment:
Can you just put the final result here and comment what it is? I'm sure it's fast enough, but the thought of it doing this math millions of times for no reason gives me the heebie jeebies.

Reply:
Oh good catch. I thought Clickhouse might evaluate that at some earlier stage but I don't think it does.